src/Sarif.Multitool/Rules/SARIF2012.ProvideRuleProperties.cs

```diff
@@ -25,19 +26,51 @@ public class ProvideHelpUris : SarifValidationSkimmerBase
         /// suppressed, noting when a violation could cause downstream tool noise, and noting when
         /// a rule can be configured in some way to refine or alter the analysis).
         /// </summary>
-        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF2012_ProvideHelpUris_FullDescription_Text };
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF2012_ProvideRuleProperties_FullDescription_Text };
 
         protected override IEnumerable<string> MessageResourceNames => new string[] {
-            nameof(RuleResources.SARIF2012_ProvideHelpUris_Note_Default_Text)
+            nameof(RuleResources.SARIF2012_ProvideRuleProperties_Note_FriendlyNameMissing_Text),
+            nameof(RuleResources.SARIF2012_ProvideRuleProperties_Note_FriendlyNameNotAPascalIdentifier_Text),
+            nameof(RuleResources.SARIF2012_ProvideRuleProperties_Note_ProvideHelpUris_Text)
         };
 
         public override FailureLevel DefaultLevel => FailureLevel.Note;
 
+        private static readonly Regex s_pascalCaseRegex = new Regex(@"^(\p{Lu}[\p{Ll}\p{Nd}]+)*$", RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

Comment:
>private [](start = 8, length = 7)

Move this down to the point where it is used. Put it directly above `Analyze(ReportingDescriptor`